components/telegram_bot_api/telegram_bot_api.app.js

```diff
@@ -89,5 +271,386 @@ module.exports = {
+     * chat. If the chat is not a private chat, the bot must be an administrator
+     * in the chat for this to work and must have the 'can_pin_messages' admin
+     * right in a supergroup or 'can_edit_messages' admin right in a channel.
+     *
+     * @param {Number|String} chatId - Unique identifier for the message
+     * recipient
+     * @param {Object} [opts] - Additional Telegram query options to be fed to
+     * the Telegram Bot API call, as defined in [the API
+     * docs](https://core.telegram.org/bots/api#unpinchatmessage)
+     * @return `True` on success
+     */
+    async unpinChatMessage(chatId, messageId) {
+      return await this.sdk().unpinChatMessage(chatId, {
+        message_id: messageId,
+      });
+    },
+    /**
+     * @typedef {function} SendMediaFn
+     * @param {string} chatId - Unique identifier for the target chat or
+     * username of the target channel (in the format @channelusername)
+     * @param {String|stream.Stream|Buffer} media - A file path or a Stream. Can
+     * also be a `file_id` previously uploaded
+     * @param {Object} [opts] - An object containing additional configuration
+     * options for this method
+     * @param {Number|String} [opts.filename] - The name of the file to send
+     * @param {Number|String} [opts.contentType] - The MIME type of the file to
+     * send
+     * @param {...*} [opts.extraOpts] - Additional Telegram query options to be
+     * fed to the Telegram Bot API call, as defined in [the API
+     * docs](https://core.telegram.org/bots/api)
+     * @returns {Promise<TelegramBot.Message>} The sent message
+    */
+    /**
+     * Send a file (Document/Image, Photo, Audio, Video, Video Note, Voice,
+     * Sticker)
+     *
+     * @param {SendMediaFn} sendFn - the function to use to send the media
+     * @param {string} chatId - Unique identifier for the target chat or
+     * username of the target channel (in the format @channelusername)
+     * @param  {String|stream.Stream|Buffer} media - A file path or a Stream. Can
+     * also be a `file_id` previously uploaded
+     * @param {Object} [opts] - An object containing additional configuration
+     * options for this method
+     * @param {Number|String} [opts.filename] - The name of the file to send
+     * @param {Number|String} [opts.contentType] - The MIME type of the file to
+     * send
+     * @param {...*} [opts.extraOpts] - Additional Telegram query options to be
+     * fed to the Telegram Bot API call, as defined in [the API
+     * docs](https://core.telegram.org/bots/api)
+     * @returns {Promise<TelegramBot.Message>} The sent message
+     */
+    async sendMedia(sendFn, chatId, media, opts) {
+      const {
+        filename,
+        contentType,
+        ...extraOpts
+      } = opts;
+      return await sendFn(chatId, media, extraOpts, {
+        filename,
+        contentType,
+      });
+    },
+    /**
+     * @typedef {import("./constants.js").UIMediaType} UIMediaType
```

Comment:
Shouldn't this be inside the jsdoc below this one?

Reply:
I think this jsdoc could be separate or part of the one below, but it might be clearer/simpler as part of the one below. I'll combine the jsdocs.